### ISSUE TITLE
Write more developer documentation (onboarding video, comments, architecture diagram)

### ISSUE THREAD

**Issue by jbilcke-hf** [jbilcke-hf/clapper#11]
### Context

As there are now code bounties, new developers might join the project but be a bit lost due to the lack of documentation.

### Goal

Today I'm pretty happy to help and answer questions in the Clapper Discord (https://discord.gg/AEruz9B92B) but it would be better to have documentation to make it scale.

So we need to work on:
- [ ] onboarding video(s)
- [ ] code documentation
- [x] diagrams

### Limitations / things to checks

T.B.D.

### How to implement

- [x] create a diagram with Excalidraw to show the interactions between the services and NPM modules

- [ ] Write a series of videos to explain each aspect of the project to developers (how screenplays are converted, data structure, the webgl timeline, rendering, AI assistant.. etc)

- [ ] Write a text document to explain a bit the architecture of the services

- [ ] Make sure all functions and components are documented (TypeScript code comments)

- [ ] Improve the README.md

### Acceptance criteria

Documentation is sufficient enough for a newcomer to run the app / do changes

**Comment by jbilcke-hf** [comment]
I've added a first version of the diagram here:

![architecture](https://media.githubusercontent.com/media/jbilcke-hf/clapper/main/documentation/diagrams/architecture-draft.png)

**Comment by lalalune** [comment]
Since the project is strongly typed in typescript, we can add easily set up an automatic workflow of typed API documentation to Docusaurus, and it will generate on CI/CD.

For example I have this project: https://github.com/jointhealliance/bgent and every time I publish, the docs publish to https://bgent.org

If we add TSDoc comments to the major functions and whatnot, it will be proper searchable API documentation unified between the codebase and the docsite.

Getting started guides and tutorials and such are important. I think that this tool will hit product market fit when you can create new tracks and clips and you don't have to leave Clapper to get your entire project done. Right now the best thing would be to get a couple more developers to help first, then it will be easy to attract more users, so getting started and dev documentation could be a good priority. However, I had no problem jumping into the project myself, it's next.js and `npm run dev`...

**Comment by jbilcke-hf** [comment]
Yes you're 100% right @lalalune , better document should help Claude figuring things out (especially meta documentation, explaining the purpose of things at a higher level than just the code, like why, interactions with other components)

I'll try to be attentive to this starting from now on

**Comment by sendn0des** [comment]
Happy to help on the onboarding videos when we're ready.
